stake-pool/js/test/instructions.test.ts

```diff
@@ -261,60 +254,31 @@ describe('StakePoolProgram', () => {
       expect(res.stakeReceiver).toEqual(undefined);
       expect(res.totalRentFreeBalances).toEqual(10000);
     });
+  });
 
-    it.only('withdraw to a stake account provided', async () => {
-      const stakeReceiver = new PublicKey(20);
-      connection.getAccountInfo = jest.fn(async (pubKey: PublicKey) => {
-        if (pubKey == stakePoolAddress) {
-          return stakePoolAccount;
-        }
-        if (pubKey.equals(CONSTANTS.poolTokenAccount)) {
-          return mockTokenAccount(LAMPORTS_PER_SOL * 2);
-        }
-        if (pubKey.equals(stakePoolMock.validatorList)) {
-          return mockValidatorList();
-        }
-        if (pubKey.equals(CONSTANTS.validatorStakeAccountAddress))
-          return mockValidatorsStakeAccount();
-        return null;
-      });
-      connection.getParsedAccountInfo = jest.fn(async (pubKey: PublicKey) => {
-        if (pubKey.equals(stakeReceiver)) {
-          return mockRpc(stakeAccountData);
-        }
-        return null;
-      });
-
-      const res = await withdrawStake(
+  describe('redelegation', () => {
+    it.only('should call successfully', async () => {
+      const data = {
         connection,
         stakePoolAddress,
-        tokenOwner,
-        1,
-        undefined,
-        undefined,
-        stakeReceiver,
+        sourceVoteAccount: PublicKey.default,
+        sourceTransientStakeSeed: 10,
+        destinationVoteAccount: PublicKey.default,
+        destinationTransientStakeSeed: 20,
+        ephemeralStakeSeed: new BN(100),
+        lamports: 100,
+      };
+      const res = await redelegate(data);
+
+      const decodedData = STAKE_POOL_INSTRUCTION_LAYOUTS.Redelegate.layout.decode(
+        res.instructions[0].data,
       );
 
-      expect((connection.getAccountInfo as jest.Mock).mock.calls.length).toBe(4);
-      expect((connection.getParsedAccountInfo as jest.Mock).mock.calls.length).toBe(1);
-      expect(res.instructions).toHaveLength(3);
-      expect(res.signers).toHaveLength(2);
-      expect(res.stakeReceiver).toEqual(stakeReceiver);
-      expect(res.totalRentFreeBalances).toEqual(10000);
-    });
-  });
-  describe('getStakeAccount', () => {
-    it.only('returns an uninitialized parsed stake account', async () => {
-      const stakeAccount = new PublicKey(20);
-      connection.getParsedAccountInfo = jest.fn(async (pubKey: PublicKey) => {
-        if (pubKey.equals(stakeAccount)) {
-          return mockRpc(uninitializedStakeAccount);
-        }
-        return null;
-      });
-      const parsedStakeAccount = await getStakeAccount(connection, stakeAccount);
-      expect((connection.getParsedAccountInfo as jest.Mock).mock.calls.length).toBe(1);
-      expect(parsedStakeAccount).toEqual(uninitializedStakeAccount.parsed);
+      expect(decodedData.instruction).toBe(19);
```

Comment:
This is currently failing:
```suggestion
      expect(decodedData.instruction).toBe(21);
```